src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java

```diff
@@ -147,19 +142,42 @@ public Mono<Either<TransactionClosureErrorEvent, TransactionClosureSentEvent>> h
                      * error event
                      */
                     // FIXME: Refactor to handle multiple notices
-                    it.pagopa.ecommerce.commons.domain.PaymentNotice paymentNotice = tx.getPaymentNotices().get(0);
+                    it.pagopa.ecommerce.commons.domain.v1.PaymentNotice paymentNotice = tx.getPaymentNotices().get(0);
                     log.info("Invoking closePaymentV2 for RptId: {}", paymentNotice.rptId().value());
                     return nodeForPspClient.closePaymentV2(closePaymentRequest)
-                            .flatMap(response -> buildEventFromOutcome(response.getOutcome(), command))
+                            .flatMap(
+                                    response -> buildClosureEvent(
+                                            command,
+                                            transactionAuthorizationCompletedData.getAuthorizationResultDto(),
+                                            response.getOutcome()
+                                    )
+                            )
                             .flatMap(transactionEventStoreRepository::save)
-                            .map(Either::<TransactionClosureErrorEvent, TransactionClosureSentEvent>right)
+                            .map(Either::<TransactionClosureErrorEvent, TransactionEvent<TransactionClosureData>>right)
                             .onErrorResume(exception -> {
-                                log.error("Got exception while invoking closePaymentV2", exception);
-                                TransactionClosureErrorEvent errorEvent = new TransactionClosureErrorEvent(
-                                        tx.getTransactionId().value().toString()
+
+                                // in case response code from Nodo is a 4xx class error retrying closure have no
+                                // meaning since it will mean that build request have an error
+                                // transactions-service side
+                                boolean unrecoverableError = exception instanceof BadGatewayException responseStatusException
+                                        && responseStatusException.getHttpStatus().is4xxClientError();
+
+                                boolean isAuthorized = AuthorizationResultDto.OK
+                                        .equals(transactionAuthorizationCompletedData.getAuthorizationResultDto());
+                                log.error(
+                                        "Got exception while invoking closePaymentV2 unrecoverable error: %s, is authorized: %s"
+                                                .formatted(unrecoverableError, isAuthorized),
+                                        exception
                                 );
+                                // the closure error event is build and sent iff the transaction was previously
```

Comment:
typo?

Reply:
If you are referring to iff here it's used as abbreviation of if and only if :)